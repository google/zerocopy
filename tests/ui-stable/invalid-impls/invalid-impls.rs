../../ui-nightly/invalid-impls/invalid-impls.rs