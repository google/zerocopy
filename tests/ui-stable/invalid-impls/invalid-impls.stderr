error[E0277]: the trait bound `T: zerocopy::FromZeroes` is not satisfied
  --> tests/ui-stable/invalid-impls/invalid-impls.rs:22:37
   |
22 | impl_or_verify!(T => FromZeroes for Foo<T>);
   |                                     ^^^^^^ the trait `zerocopy::FromZeroes` is not implemented for `T`
   |
note: required for `Foo<T>` to implement `zerocopy::FromZeroes`
  --> tests/ui-stable/invalid-impls/invalid-impls.rs:18:10
   |
18 | #[derive(FromZeroes, FromBytes, AsBytes, Unaligned)]
   |          ^^^^^^^^^^ unsatisfied trait bound introduced in this `derive` macro
note: required by a bound in `_::Subtrait`
  --> tests/ui-stable/invalid-impls/../../../src/macros.rs
   |
   |             trait Subtrait: $trait {}
   |                             ^^^^^^ required by this bound in `Subtrait`
   |
  ::: tests/ui-stable/invalid-impls/invalid-impls.rs:22:1
   |
22 | impl_or_verify!(T => FromZeroes for Foo<T>);
   | ------------------------------------------- in this macro invocation
   = note: this error originates in the derive macro `FromZeroes` which comes from the expansion of the macro `impl_or_verify` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T`
    |
22  | impl_or_verify!(T: zerocopy::FromZeroes => FromZeroes for Foo<T>);
    |                  ++++++++++++++++++++++

error[E0277]: the trait bound `T: zerocopy::FromBytes` is not satisfied
  --> tests/ui-stable/invalid-impls/invalid-impls.rs:23:36
   |
23 | impl_or_verify!(T => FromBytes for Foo<T>);
   |                                    ^^^^^^ the trait `zerocopy::FromBytes` is not implemented for `T`
   |
note: required for `Foo<T>` to implement `zerocopy::FromBytes`
  --> tests/ui-stable/invalid-impls/invalid-impls.rs:18:22
   |
18 | #[derive(FromZeroes, FromBytes, AsBytes, Unaligned)]
   |                      ^^^^^^^^^ unsatisfied trait bound introduced in this `derive` macro
note: required by a bound in `_::Subtrait`
  --> tests/ui-stable/invalid-impls/../../../src/macros.rs
   |
   |             trait Subtrait: $trait {}
   |                             ^^^^^^ required by this bound in `Subtrait`
   |
  ::: tests/ui-stable/invalid-impls/invalid-impls.rs:23:1
   |
23 | impl_or_verify!(T => FromBytes for Foo<T>);
   | ------------------------------------------ in this macro invocation
   = note: this error originates in the derive macro `FromBytes` which comes from the expansion of the macro `impl_or_verify` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T`
    |
23  | impl_or_verify!(T: zerocopy::FromBytes => FromBytes for Foo<T>);
    |                  +++++++++++++++++++++

error[E0277]: the trait bound `T: zerocopy::AsBytes` is not satisfied
  --> tests/ui-stable/invalid-impls/invalid-impls.rs:24:34
   |
24 | impl_or_verify!(T => AsBytes for Foo<T>);
   |                                  ^^^^^^ the trait `zerocopy::AsBytes` is not implemented for `T`
   |
note: required for `Foo<T>` to implement `zerocopy::AsBytes`
  --> tests/ui-stable/invalid-impls/invalid-impls.rs:18:33
   |
18 | #[derive(FromZeroes, FromBytes, AsBytes, Unaligned)]
   |                                 ^^^^^^^ unsatisfied trait bound introduced in this `derive` macro
note: required by a bound in `_::Subtrait`
  --> tests/ui-stable/invalid-impls/../../../src/macros.rs
   |
   |             trait Subtrait: $trait {}
   |                             ^^^^^^ required by this bound in `Subtrait`
   |
  ::: tests/ui-stable/invalid-impls/invalid-impls.rs:24:1
   |
24 | impl_or_verify!(T => AsBytes for Foo<T>);
   | ---------------------------------------- in this macro invocation
   = note: this error originates in the derive macro `AsBytes` which comes from the expansion of the macro `impl_or_verify` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T`
    |
24  | impl_or_verify!(T: zerocopy::AsBytes => AsBytes for Foo<T>);
    |                  +++++++++++++++++++

error[E0277]: the trait bound `T: zerocopy::Unaligned` is not satisfied
  --> tests/ui-stable/invalid-impls/invalid-impls.rs:25:36
   |
25 | impl_or_verify!(T => Unaligned for Foo<T>);
   |                                    ^^^^^^ the trait `zerocopy::Unaligned` is not implemented for `T`
   |
note: required for `Foo<T>` to implement `zerocopy::Unaligned`
  --> tests/ui-stable/invalid-impls/invalid-impls.rs:18:42
   |
18 | #[derive(FromZeroes, FromBytes, AsBytes, Unaligned)]
   |                                          ^^^^^^^^^ unsatisfied trait bound introduced in this `derive` macro
note: required by a bound in `_::Subtrait`
  --> tests/ui-stable/invalid-impls/../../../src/macros.rs
   |
   |             trait Subtrait: $trait {}
   |                             ^^^^^^ required by this bound in `Subtrait`
   |
  ::: tests/ui-stable/invalid-impls/invalid-impls.rs:25:1
   |
25 | impl_or_verify!(T => Unaligned for Foo<T>);
   | ------------------------------------------ in this macro invocation
   = note: this error originates in the derive macro `Unaligned` which comes from the expansion of the macro `impl_or_verify` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T`
    |
25  | impl_or_verify!(T: zerocopy::Unaligned => Unaligned for Foo<T>);
    |                  +++++++++++++++++++++
